core/domain/exp_services.py

```diff
@@ -1362,15 +1380,45 @@ def _save_exploration(
 
     # Update the version history data for each state and the exploration
     # metadata in the new version of the exploration.
-    update_version_history(
-        exploration, change_list, committer_id, old_states, old_metadata)
+    models_to_put.append(
+        get_updated_version_history_model(
+            exploration,
+            change_list,
+            committer_id,
+            old_states,
+            old_metadata
+        )
+    )
 
     # Trigger statistics model update.
     new_exp_stats = stats_services.get_stats_for_new_exp_version(
         exploration.id, exploration.version, list(exploration.states.keys()),
-        exp_versions_diff, None)
+        exp_versions_diff, revert_to_version=None)
 
-    stats_services.create_stats_model(new_exp_stats)
+    new_state_stats_mapping = {
```

Comment:
Same thing, no copy-paste please. Refactor to deduplicate. In this case, create a new function for returning the new_state_stats mapping and have this and the create_stats_model() functions call it.